File: app/src/main/java/fi/kaupunkifillarit/analytics/InfoDrawerEvents.java
package fi.kaupunkifillarit.analytics;

import com.google.android.gms.analytics.HitBuilders;

import java.util.Map;

public abstract class InfoDrawerEvents {
    private static final String CATEGORY = "Info Drawer";

    public static Map<String, String> open() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_OPEN)
                .setValue(1)
                .build();
    }

    public static Map<String, String> close() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLOSE)
                .setValue(1)
                .build();
    }

    public static Map<String, String> shareClick() {
        return new HitBuilders.EventBuilder()
                .setCategory(Events.CATEGORY_SHARE)
                .setAction(Events.ACTION_CLICK)
                .setValue(1)
                .build();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/analytics/InfoDrawerEvents.kt
package fi.kaupunkifillarit.analytics

import com.google.android.gms.analytics.HitBuilders

object InfoDrawerEvents {
    private const val CATEGORY = "Info Drawer"

    fun open(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_OPEN)
                .setValue(1)
                .build()
    }

    fun close(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLOSE)
                .setValue(1)
                .build()
    }

    fun shareClick(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(Events.CATEGORY_SHARE)
                .setAction(Events.ACTION_CLICK)
                .setValue(1)
                .build()
    }
}
